compiler.ts

```diff
@@ -289,4 +290,373 @@ function codeGenClass(cls : Class<[Type, SourceLocation]>, env : GlobalEnv) : Ar
   methods.forEach(method => method.name = `${cls.name}$${method.name}`);
   const result = methods.map(method => codeGenDef(method, env));
   return result.flat();
-  }
+}
+
+function setFuncs() : Array<string> {
```

Comment:
Defining set functions like this is not a great idea. This is extremely hard to test. It's located in compiler which suggests somebody using compiler should be aware of this. This just doesn't belong here and it should probably be written in a separate, testable WASM module.

Reply:
I tried to put them in a separate file called sets.wat under the /stdlib folder, and build them like we did to memory.wat into sets.wasm. But I encountered some problems when loading it into the code. For now, I moved the set-related code into memory.wat and load them together with the original three memory-related function, and things worked out okay. We'd appreciate it if you could help us find out what was wrong with our first approach.

Reply:
Update: This is fixed. We put all the set-related wasm code in build/sets.wat, and build them into sets.wasm. We load these functions separately into our program.